### PR TITLE
PLUGIN-1578 : Confluent Kafka Streaming Source Plugin | Support Confluent Platform

### PR DESCRIPTION
**Pull request by sgarg-CS**
[PLUGIN-1578](https://cdap.atlassian.net/browse/PLUGIN-1578)

### CODE REVIEW

File: confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceUtil.java [hunk 1]
```diff
@@ -215,11 +215,14 @@ private static Map<String, Object> getConsumerParams(ConfluentStreamingSourceCon
     // Spark saves the offsets in checkpoints, no need for Kafka to save them
     kafkaParams.put(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "https");
     kafkaParams.put(CommonClientConfigs.RETRY_BACKOFF_MS_CONFIG, "500");
-    kafkaParams.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
-    kafkaParams.put(SaslConfigs.SASL_MECHANISM, "PLAIN");
-    kafkaParams.put(SaslConfigs.SASL_JAAS_CONFIG, "org.apache.kafka.common.security.plain.PlainLoginModule required " +
-      "username=\"" + conf.getClusterApiKey() + "\" password=\"" + conf.getClusterApiSecret() + "\";");
 
+    if (!Strings.isNullOrEmpty(conf.getClusterApiKey()) || !Strings.isNullOrEmpty(conf.getClusterApiSecret())) {
```
Comment (greeshmaswaminathan) [inline confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceUtil.java]: Does validation work with just api key and not secret? If both are required, use `&&` . Consider adding validation if only one value is present. Can we also update the tooltips for these fields to indicate that they are required with confluent cloud?

Comment (sgarg-CS) [reply confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceUtil.java]: No, both are required. Changed to `&&`
Added validation if only one value is present. 
Also updated the tooltip.
22d6525bd191735661c4b0d8f51208b14365000f

File: confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceConfig.java [hunk 2]
```diff
@@ -161,13 +161,15 @@ public class ConfluentStreamingSourceConfig extends ReferencePluginConfig implem
   private final String kafkaProperties;
 
   @Name(NAME_CLUSTER_API_KEY)
-  @Description("The Confluent API Key.")
+  @Description("The Confluent API Key. Required if connecting with Confluent Cloud")
```
Comment (greeshmaswaminathan) [inline confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceConfig.java]: nit: Add a `.` at the end for both descriptions.

Comment (greeshmaswaminathan) [reply confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceConfig.java]: Do we also need to update .md files ?

Comment (sgarg-CS) [reply confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceConfig.java]: Added `.` at the end for both descriptions. Also updated the `.md` file. e6a2fc04c73eb0bf81df50477d75daa995e44f6f

**Comment by sgarg-CS** [comment]
Squashed the commits. e528d858588f9b014072d8305493395b773ebab8 Thank you!

File: confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.confluent.streaming.source.ConfluentStreamingSourceConfig;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConfluentStreamingSourceConfigTest {
```
Comment (greeshmaswaminathan) [inline confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigTest.java:30]: Add java doc.

File: confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigHelper.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    public ConfluentStreamingSourceConfig build() {
+      return new ConfluentStreamingSourceConfig(referenceName, brokers, topic, partitions, null,
+        null, schema, format, null, null, null, null,
+        maxRatePerPartition, null, clusterApiKey, clusterApiSecret, null,
+        null, null, null);
+    }
+  }
+
```
Comment (greeshmaswaminathan) [inline confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigHelper.java]: nit: Remove new line here.

File: confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigTest.java (hunk 3 above)
Comment (sgarg-CS) [reply confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigTest.java:30]: Added Javadoc for ConfluentStreamingSourceConfigTest class 54ff8ddf31cfc9ea820e08bcd92349669b88f59a

File: confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigHelper.java (hunk 4 above)
Comment (sgarg-CS) [reply confluent-kafka-plugins/src/test/java/io/cdap/plugin/confluent/integration/streaming/source/ConfluentStreamingSourceConfigHelper.java]: Removed. 54ff8ddf31cfc9ea820e08bcd92349669b88f59a

**Comment by sgarg-CS** [comment]
Commits have been squashed.

**Review by greeshmaswaminathan — APPROVED** [review]

**Pull request merged**